### PR TITLE
[WFLY-5873, JBEAP-1453] Can't set static-connectors on AMQ bridge

### PR DESCRIPTION
**Pull request by TomasHofman**
... if discovery-group was set before

https://issues.jboss.org/browse/WFLY-5873
https://issues.jboss.org/browse/JBEAP-1453

### CODE REVIEW

**Comment by TomasHofman** [comment]
I'm investigating the failures.

**Comment by TomasHofman** [comment]
Retest this please.

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java [hunk 1]
```diff
@@ -177,9 +177,6 @@
             .setAllowNull(true)
             // do not allow expression as this may reference another resource
             .setAllowExpression(false)
-            .setAlternatives("socket-binding",
-                    "group-address", "group-port",
-                    "local-bind-address", "local-bind-port")
```
Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java]: SOCKET_BINDING lists this JGROUPS_STACK as an alternative.

**Comment by bstansberry** [comment]
@jmesnil Please approve. Looks ok to me except I don't know about the alternatives for JGROUPS_STACK.

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java (hunk 1 above)
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java]: @TomasHofman you can also remove `"group-address", "group-port", "local-bind-address", "local-bind-port"`from the list of alternatives. They were valid in the legacy messaging subsystem but they no longer exist in the new messaging-activemq subsystem.

**Comment by jmesnil** [comment]
`socket-binding` and `jgroups-channel` are alternatives of each other.
`jgroups-stack` is an optional attribute that is required only when `jgroups-channel` is used (and if `jgroups-stack` is not defined, it used the default stack defined in the `jgroups` subsystem).

The original idea to flag `socket-binding` as an alternative to `jgroups-stack` was to prevent having the user define both `socket-binding` and `jgroups-stack`.

@bstansberry Does that make sense? We could also remove `jgroups-stack` from the alternatives of `socket-binding`. Defining both is not correct from a conceptual pov but it will not create any issue as the `jgroups-stack` is not used unless `jgroups-channel` is defined.

**Comment by bstansberry** [comment]
@jmesnil @TomasHofman based on the spec at https://docs.jboss.org/author/display/WFLY10/Admin+Guide#AdminGuide-DescriptionofanAttribute I think what is appropriate here is that JGROUPS_STACK and SOCKET_BINDING do not include each other in their respective setAlternatives(...) list, but JGROUPS_STACK should include the following:

```
.setRequires("jgroups-channel")
```

This indicates that use of jgroups-stack indicates an intent to configure a jgroups-channel.

If you guys prefer though, I'm fine though with merging this PR as is, if all else is ok with you Jeff, and then the setRequires stuff can come later. I don't want to hold this up.

**Comment by TomasHofman** [comment]
True... I will fix the PR according to Brian's suggestion, unless you disagree @jmesnil

**Comment by jmesnil** [comment]
@TomasHofman yes, please follow brian's suggestion so that we clean up these attributes once and for all. thanks!

**Comment by TomasHofman** [comment]
Just updated PR, thanks for advice @bstansberry

**Comment by TomasHofman** [comment]
retest this please

**Comment by TomasHofman** [comment]
retest this please

**Comment by TomasHofman** [comment]
retest this please

**Comment by TomasHofman** [comment]
retest this please

**Comment by n1hility** [comment]
@jmesnil can you approve the final version?

**Comment by jmesnil** [comment]
@n1hility this looks good.

**Pull request merged**